### PR TITLE
All functionality for the admin program is now located in admin.go

### PR DESCRIPTION
**Pull request by ocket8888**
## What does this PR do?
Makes the deprecated `admin.pl` a small wrapper around the `admin` binary, so that new functionality need only be added in one place.

## Which TC components are affected by this PR?

- [ ] Documentation
- [ ] Grove
- [ ] Traffic Analytics
- [ ] Traffic Monitor
- [ ] Traffic Ops
- [ ] Traffic Ops ORT
- [ ] Traffic Portal
- [ ] Traffic Router
- [ ] Traffic Stats
- [ ] Traffic Vault
- [x] Other: admin.pl

## What is the best way to verify this PR?
Run any and all tests for the `admin.pl` script

## Check all that apply

- [ ] This PR includes tests
- [ ] This PR includes documentation updates
- [ ] This PR includes an update to CHANGELOG.md
- [ ] This PR includes all required license headers
- [ ] This PR includes a database migration (ensure that migration sequence is correct)
- [ ] This PR fixes a serious security flaw. Read more: [www.apache.org/security](http://www.apache.org/security/)

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/3011/
Test PASSed.

File: traffic_ops/app/db/admin.pl [hunk 1]
```diff
@@ -14,261 +14,8 @@
...
-	}
-}
-
-sub show_users {
-	if ( system("psql -h $host_ip -p $host_port -U $db_super_user -ec '\\du';") != 0 ) {
-		die "Can't show users";
-	}
-}
-
-sub reverse_schema {
```
Comment (rawlinp) [inline traffic_ops/app/db/admin.pl]: Shoot, I just remembered that this function is still needed as long as we have Perl TO API endpoints, and the admin binary actually shells out this command to this admin.pl script. So this function should probably be pulled out into its own little Perl script that the admin binary can shell that command out to. Then admin.pl can just be as you have it now. Otherwise, in this PR if you try running this command, I think it will just create an endless cycle due to the circular dependency.

Comment (ocket8888) [reply traffic_ops/app/db/admin.pl]: shouldn't that be implemented in the binary?

Comment (ocket8888) [reply traffic_ops/app/db/admin.pl]: I'm concerned though - because this needs to exist as long as the Perl back-end exists, and because we deprecated the script, it sort of seems like we're now locked in to a release schedule that demands that 4.0 not be released until the entire Perl TO codebase can be removed.

Comment (rob05c) [reply traffic_ops/app/db/admin.pl]: Can we put `reverse_schema` in its own script file, `admin_reverse_schema.pl`, with only that function, and make Go execute that script?

It's not ideal, but the only dependencies it will have is DBIx, which fortunately is in a CentOS package, so it shouldn't need CPan or Carton.

Comment (ocket8888) [reply traffic_ops/app/db/admin.pl]: We could, but idk how much that really buys us. It also has a dependency on `Schema` btw, but I think that's maybe a standard library thing? Not sure. Leaving it in `admin.pl` means adding two dependencies: `GetOpt::Long` (standard?) and `YAML`, which I don't personally think of as a huge deal, but it is two places to maintain the command line argument format. On the other hand separating it means we now have three programs to maintain for the `admin` tool, at least until `admin.pl` vanishes. It's also functionality specific to `admin.pl` since it only deals with Perl data representation and should vanish with the need for that script

In summary, I'm sort of torn on the idea of putting that in a new script, and I'm leaning towards saying that that's something that should maybe take place once `admin.pl` actually gets removed (assuming the Perl back-end still exists at that point).

Comment (rob05c) [reply traffic_ops/app/db/admin.pl]: >It also has a dependency on Schema 
`Schema` is DBIx.

>two dependencies: GetOpt::Long (standard?) and YAML

>two places to maintain the command line argument format

I'm not suggesting maintaining the argument format. Make the script take the arguments it needs (environment, user, pass, host, port), and make Go pass them. No YAML or GetOpt.

Comment (ocket8888) [reply traffic_ops/app/db/admin.pl]: I think you misunderstood, those dependencies and that maintenance *only apply* if we're leaving it in `admin.pl`, not if we put it in a separate script as you suggest.

Comment (rawlinp) [reply traffic_ops/app/db/admin.pl]: We shouldn't really be worried about which dependencies the new reverse_schema.pl script will require, since as long as we have Perl endpoints we will still have the Perl dependencies required for reverse_schema. For now, yes, we will end up with 3 programs to maintain until we get rid of admin.pl, but that shouldn't really be a concern either because reverse_schema.pl will only contain that required functionality and won't require any further updates until it's eventually removed with TO-Perl. And with admin.pl just being a thin Perl wrapper over the admin.go binary, there really won't be much to maintain there either.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/3018/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/3019/
Test FAILed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/3020/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/3021/
Test PASSed.

File: traffic_ops/app/db/admin.pl (hunk 1 above)
Comment (ocket8888) [reply traffic_ops/app/db/admin.pl]: the alternative I'm suggesting is keep that one function in `admin.pl` and remove everything else. It's not as clean as making a new script, I guess, but that's not super clean either. I suppose I just don't see the advantage of moving this to a new file until we're prepared to get rid of `admin.pl` in 4.0.

Comment (rob05c) [reply traffic_ops/app/db/admin.pl]: As long as `admin.pl` is kept as an empty wrapper, having real functionality called by `admin.go` creates a circular dependency. It can work, but circular scripts are perilous. Much better, simpler, safer, easier to understand, to make a third script to avoid the cycle.

Comment (ocket8888) [reply traffic_ops/app/db/admin.pl]: Well I was going to have `admin.pl` intercept the `reverse_schema` functionality such that it never calls `admin` in that case. No circularity required.

Comment (rob05c) [reply traffic_ops/app/db/admin.pl]: But `admin.go` has to call `admin.pl` for that function as well. The scripts are still calling each-other. Even if the code execution isn't circular, the scripts are. That's confusing and dangerous.

Comment (ocket8888) [reply traffic_ops/app/db/admin.pl]: Well in any case I don't feel strongly about it. I can separate it into its own script.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/3022/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/3025/
Test PASSed.

File: traffic_ops/app/db/admin.pl (hunk 1 above)
Comment (rawlinp) [reply traffic_ops/app/db/admin.pl]: The `get_dbinfo` function checks the environment variables for MOJO_MODE and uses that to determine which database.conf to read the config from. By default it will use conf/development/database.conf. That comes from traffic_ops/app/lib/Schema.pm I believe.

I'll pull this down and take a spin.

File: traffic_ops/app/db/admin.go [hunk 2]
```diff
@@ -303,12 +303,12 @@ func loadSchema() {
 }
 
 func reverseSchema() {
-	fmt.Println("WARNING: the '" + CmdReverseSchema + "' command will be removed with Traffic Ops Perl because it will no longer be necessary")
-	cmd := exec.Command("db/admin.pl", "--env="+Environment, CmdReverseSchema)
+	fmt.Fprintf(os.Stderr, "WARNING: the '%s' command will be removed with Traffic Ops Perl because it will no longer be necessary\n", CmdReverseSchema)
+	cmd := exec.Command("db/reverse_schema.pl")
```
Comment (rawlinp) [inline traffic_ops/app/db/admin.go]: This command needs the MOJO_MODE environment variable injected in order to reverse_schema on a specific environment (rather than just "development"):
```
cmd := exec.Command("db/reverse_schema.pl")
cmd.Env = append(os.Environ(), "MOJO_MODE="+Environment)
...
```

With that fixed I believe this will be good to go.

Comment (ocket8888) [reply traffic_ops/app/db/admin.go]: Really? Wouldn't it inherit the environment of the parent process? It seemed to be working for me, but if you think it needs this I'll add it.

Comment (ocket8888) [reply traffic_ops/app/db/admin.go]: From the Go docs:
```go
        // If Env is nil, the new process uses the current process's
        // environment.
```
However, it looks like the re-write to Go dropped the Perl script's behaviour of setting `MOJO_MODE` - probably because only `reverse_schema` used it and that was calling back to the Perl script anyway.

tl;dr I took your suggestion.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/3031/
Test PASSed.
